File: src/main/java/com/fabledzx/MyFirstMod.java
package com.fabledzx;

import net.fabricmc.api.ModInitializer;

public class MyFirstMod implements ModInitializer {
    public static final String MOD_ID = "myfirstmod";

    @Override
    public void onInitialize() {
        ModItems.register();
        ModItems.register1();
        System.out.println("Hello Fabric World!");
    }
}

File: src/main/kotlin/com/fabledzx/MyFirstMod.kt
package com.fabledzx

import net.fabricmc.api.ModInitializer
import net.minecraft.util.Identifier
import org.slf4j.LoggerFactory

object MyFirstMod : ModInitializer {
	const val MOD_ID: String = "myfirstmod"

	private val LOGGER = LoggerFactory.getLogger(MOD_ID)

	override fun onInitialize() {
		// This code runs as soon as Minecraft is in a mod-load-ready state.
		// However, some things (like resources) may still be uninitialized.
		// Proceed with mild caution.

		LOGGER.info("Hello Fabric world!")
	}

	fun id(path: String): Identifier
		= new Identifier(MOD_ID, path)
}
